airbrakes/data_handling/data_processor.py

```diff
@@ -160,15 +164,32 @@ def get_processed_data(self) -> list[ProcessedDataPacket]:
 
         :return: A list of ProcessedDataPacket objects.
         """
+        # Because estPressureAlt only updates every ~20ms, we need to "fill in" the speeds derived from it
+        # to match the length of the speeds derived from acceleration--we do this via stepwise interpolation
+        # _current_altitudes and _speeds_from_altitude should have the same length
+        speeds_from_altitude_interpolated = []
+        for i in range(len(self._current_altitudes)):
+            # If the altitude has changed, we need to interpolate the speed, but you get n - 1 speeds from n altitudes,
+            # so we need to check that we have speeds left
+
+            if self._current_altitudes[i] != self._previous_altitude and len(self._speeds_from_altitude) != 0:
+                self._previous_altitude = self._current_altitudes[i]
+                self._previous_velocity_from_altitude = self._speeds_from_altitude[0]
+                self._speeds_from_altitude = self._speeds_from_altitude[1:]
+            speeds_from_altitude_interpolated.append(self._previous_velocity_from_altitude)
```

Comment:
I suspected that there was a numpy function for interpolation so I asked chatgpt and it looks like there is, not sure if you've looked at that already:

```python
import numpy as np

# Example altitudes and speeds (make sure to replace with your actual data)
altitudes = np.array(self._current_altitudes)
speeds = np.array(self._speeds_from_altitude)

# Interpolating the speeds for the entire altitudes range
speeds_interpolated = np.interp(altitudes, self._previous_altitudes, self._previous_speeds_from_altitude)

# If you need to append the interpolated speeds to a list or similar structure
speeds_from_altitude_interpolated = list(speeds_interpolated)
```

Reply:
that linearly interpolates rather than doing it in steps. Both are fine as we will just be using the actual values in the kalman filter, not the interpolated ones. I feel like stepwise interpolation is better tho as it is more accurate to what is actually happening